src/snowflake/snowpark/_internal/server_connection.py

```diff
@@ -452,10 +452,11 @@ def run_batch_insert(self, query: str, rows: List[Row], **kwargs) -> None:
         # with qmark, Python data type will be dynamically mapped to Snowflake data type
         # https://docs.snowflake.com/en/user-guide/python-connector-api.html#data-type-mappings-for-qmark-and-numeric-bindings
         params = [list(row) for row in rows]
+        statement_params = kwargs.get("_statement_params")
         query_tag = (
-            kwargs["_statement_params"]["QUERY_TAG"]
-            if "_statement_params" in kwargs
-            and "QUERY_TAG" in kwargs["_statement_params"]
+            statement_params["QUERY_TAG"]
+            if statement_params is not None
+            and "QUERY_TAG" in statement_params
```

Comment:
Irrelevant to this PR. When `save_as_table` is called against a DataFrame with "large local data", an exception is thrown there because `_statement_params ` is in the kwargs but value is `None`. This is to fix it so a test in this PR can be passed.

Reply:
Just curious: Did we explicitly set `_statement_params` to `None` somewhere?